### PR TITLE
feat: update burn transaction for core key manager

### PR DESCRIPTION
**Pull request by hansieodendaal**
Description
---
- Updated burn transaction to make use of the core key manager
- Updated required infrastructure and tests to facilitate this
- Added a unit test for burn tari that utilises the new core key manager functions

Motivation and Context
---
Part of the key manager refactoring

How Has This Been Tested?
---
Unit tests and integration tests

What process can a PR reviewer use to test or verify this change?
---
See unit test `async fn single_transaction_burn_tari()`

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

File: base_layer/core/src/core_key_manager/handle.rs [hunk 1]
```diff
@@ -116,6 +118,14 @@ where TBackend: KeyManagerBackend<PublicKey> + 'static
             .await
     }
 
+    async fn get_static_key_id<T: Into<String> + Send>(&self, branch: T) -> Result<KeyId, KeyManagerServiceError> {
+        (*self.core_key_manager_inner)
+            .read()
+            .await
+            .get_next_key_id(&branch.into())
```
Comment (SWvheerden) [inline base_layer/core/src/core_key_manager/handle.rs]: this seems incorrect

File: base_layer/core/src/test_helpers/mod.rs [hunk 2]
```diff
@@ -54,13 +54,13 @@ use crate::{
 #[macro_use]
 mod block_spec;
 pub mod blockchain;
-pub type TestKeyManager = CoreKeyManagerHandle<KeyManagerSqliteDatabase<DbConnection>>;
+pub type MemoryDbCoreKeyManagerHandle = CoreKeyManagerHandle<KeyManagerSqliteDatabase<DbConnection>>;
```
Comment (SWvheerden) [inline base_layer/core/src/test_helpers/mod.rs]: I much prefer the older name, this name might infer the proper type, but for 99% of tests, just seeing testkeymanager as the type is much simpler to read and see what you use it for

File: base_layer/key_manager/src/key_manager_service/handle.rs [hunk 3]
```diff
@@ -88,8 +88,12 @@ where
             .await
     }
 
-    async fn get_next_key_id<T: Into<String> + Send>(&self, _branch: T) -> Result<KeyId, KeyManagerServiceError> {
-        todo!()
+    async fn get_next_key_id<T: Into<String> + Send>(&self, branch: T) -> Result<KeyId, KeyManagerServiceError> {
+        unimplemented!("KeyManagerHandle::get_next_key_id({})", branch.into())
```
Comment (SWvheerden) [inline base_layer/key_manager/src/key_manager_service/handle.rs]: I would leave this todo as well, we should implement this before merging to dev

File: base_layer/wallet/src/output_manager_service/handle.rs [hunk 4]
```diff
@@ -136,11 +138,17 @@ pub enum OutputManagerRequest {
...
     GetOutputStatusesByTxId(TxId),
-    GetNextSpendAndScriptKeys,
-    GetRecoveryData,
+    GetNextSpendAndScriptKeys, // TODO: Remove this when core key manager is implemented
+    GetNextSpendAndScriptKeyIds,
+    GetRecoveryData, // TODO: Remove this when core key manager is implemented
+    GetRecoveryKeyId,
+    TryCommitmentKeyRecovery(Commitment, EncryptedData, Option<KeyId>),
+    GetSpendingKeyId(PublicKey),
+    GetDiffieHellmanSharedSecret(KeyId, PublicKey),
```
Comment (SWvheerden) [inline base_layer/wallet/src/output_manager_service/handle.rs]: we can remove all of these we don't need them

Comment (hansieodendaal) [reply base_layer/wallet/src/output_manager_service/handle.rs]: I will verify and remove, together with their downline implementations

Comment (hansieodendaal) [reply base_layer/wallet/src/output_manager_service/handle.rs]: I have removed `GetNextSpendAndScriptKeys`, `GetRecoveryData` and `GetDiffieHellmanSharedSecret` for now

Comment (hansieodendaal) [reply base_layer/wallet/src/output_manager_service/handle.rs]: Ok, removed all of them

File: base_layer/core/src/core_key_manager/interface.rs [hunk 5]
```diff
@@ -51,6 +56,7 @@ impl CoreKeyManagerBranch {
             CoreKeyManagerBranch::CommitmentMask => "core: commitment mask".to_string(),
             CoreKeyManagerBranch::Nonce => "core: nonce".to_string(),
             CoreKeyManagerBranch::ScriptKey => "core: script key".to_string(),
+            CoreKeyManagerBranch::Recovery => "core: recovery".to_string(),
```
Comment (SWvheerden) [inline base_layer/core/src/core_key_manager/interface.rs:59]: we need to chose data_encryption or recovery here.

**Review by SWvheerden — APPROVED** [review]

**Pull request merged**